docs/how-to-integrate-a-frontend.md

```diff
@@ -0,0 +1,247 @@
+
+A `Provider` is an abstraction for a Bitcoin node that you connect with when you need to interact with the blockchain, e.g., to broadcast a transaction. Whatsonchain is an example, providing access to the blockchain.
+
+### Connect Sensilet Wallet
+
+After clicking the **Connect Sensilet** button, we initialize a `SensiletSigner` and save it.
+
+In the `App.tsx` file, we then call the `getDefaultPubKey()` interface of the wallet to request to connect to get the public keys of the two players.
+
+```ts
+const sensiletLogin = async () => {
+    try {
+      const provider = new DefaultProvider();
+      const signer = new SensiletSigner(provider);
+
+      signerRef.current = signer;
+      
+      setConnected(true);
+
+      const alicPubkey = await signer.getDefaultPubKey();
+      setAlicePubkey(toHex(alicPubkey))
+
+      // Prompt user to switch accounts
+
+    } catch (error) {
+      console.error("sensiletLogin failed", error);
+      alert("sensiletLogin failed")
+    }
+};
+```
+
+### Show Balance
+
+After the connection is successful, we all the `getBalance()` interface to get the wallet balance:
+
+```ts
+signer.getBalance().then(balance => 
+  // UTXOs belonging to transactions in the mempool are unconfirmed
+  setAliceBalance(balance.confirmed + balance.unconfirmed)
+);
+```
+
+If a wallet is connected, its balance is displayed. Otherwise, we show the connect button.
+
+```ts
+{
+  isConnected ?
+    <div>
+      <label>Alice Balance: {alicebalance} <span> (satoshis)</span></label>
+    </div>
+    :
+    <button
+      className="pure-button button-large sensilet"
+      onClick={sensiletLogin}
+    >
+      Connect Wallet
+    </button>
+}
+```
+
+![](../static/img/connect.gif)
+
+## Initialize Contract
+
+We have obtained the contract class `TicTacToe` by loading the contract artifact file. When the user clicks the start button, the contract is initialized with the public keys of two players `alice` and `bob`. The public key can be obtained through the `getDefaultPubKey()` interface of `Signer`.
+
+```ts
+const signer = signerRef.current as SensiletSigner;
+
+const instance = new TicTacToe(
+    PubKey(toHex(alicePubkey)),
+    PubKey(toHex(bobPubkey))
+);
+
+await instance.connect(signer);
+```
+
+## Deploy Contract
+
+Each contract instance has a `deploy` method:
+
+```ts
+async deploy(amount:number = 1, options?: { 
+    changeAddress?: AddressOption,
+    address?: AddressOption
+}): Promise<TransactionResponse>;
+```
+
+- `amount`: how many satoshis would be locked in the contract when deployed.
+- `options`: an optional parameter, `changeAddress` is a customed change address, and the corresponding private key of `address` will be used to sign the transaction.
+
```

Comment:
Just say you can deploy & call contract as in https://scrypt.io/scrypt-ts/how-to-deploy-and-call-a-contract#contract-deployment. No need to repeat if it's the same as in nodejs env.